src/Microsoft.DotNet.Wpf/src/PresentationFramework/System/Windows/ThemeManager3.cs

```diff
@@ -0,0 +1,434 @@
+
+        foreach(int index in indices)
+        {
+            Application.Current.Resources.MergedDictionaries.RemoveAt(index);
+        }
+
+        foreach(Window window in Application.Current.Windows)
+        {
+            if(!FluentEnabledWindows.HasItem(window))
+            {
+                RemoveStyleFromWindow(window);
+            }
+        }
+    }
+
+    private static void RemoveFluentFromWindow(Window window)
+    {
+        if(window == null || window.IsDisposed) return;
+
+        List<int> indices = FindAllFluentThemeResourceDictionaryIndex(Application.Current.Resources);
+
+        foreach(int index in indices)
+        {
+            Application.Current.Resources.MergedDictionaries.RemoveAt(index);
+        }
+
+        RemoveStyleFromWindow(window);
+        FluentEnabledWindows.Remove(window);
+    }
+
+    private static void ApplyFluentOnWindow(Window window)
+    {
+        if(window == null || window.IsDisposed) return;
+        
+        bool useLightColors = GetUseLightColors(window.ThemeMode);
+        var fluentThemeResourceUri = GetFluentThemeResourceUri(useLightColors);
+        AddOrUpdateThemeResources(window.Resources, fluentThemeResourceUri);
+        ApplyStyleOnWindow(window, useLightColors);
+
+        if(!FluentEnabledWindows.HasItem(window))
+        {
+            FluentEnabledWindows.Add(window);
+        }
+    }
+
+    private static void RemoveStyleFromWindow(Window window)
+    {
+        if(window == null || window.IsDisposed) return;
+
+        if(IsFluentThemeEnabled || window.ThemeMode != ThemeMode.None)
+        {
+            bool useLightColors = GetUseLightColors(Application.Current.ThemeMode);
+            window.SetImmersiveDarkMode(!useLightColors);
+        }
+        else
+        {
+            // TODO : Remove the styles from windows which have BackdropDisabledWidowStyle
+            window.SetImmersiveDarkMode(false);
+            WindowBackdropManager.SetBackdrop(window, WindowBackdropType.None);
+        }
+
+    }
+
+    private static void ApplyStyleOnWindow(Window window, bool useLightColors)
+    {
+        if(window == null || window.IsDisposed) return;
+
+        // Add a check to see if the window already has a non-implicit style set.
+        window.SetResourceReference(FrameworkElement.StyleProperty, typeof(Window));
+        window.SetImmersiveDarkMode(!useLightColors);
+
+        if(SystemParameters.HighContrast)
+        {
+            WindowBackdropManager.SetBackdrop(window, WindowBackdropType.None);
+        }
+        else
+        {
+            WindowBackdropManager.SetBackdrop(window, WindowBackdropType.MainWindow);
+        }
+    }
+
+    #endregion
+
+
+    #region Internal Properties
+
+    internal static bool DeferSyncingThemeModeAndResources { get; set; } = true;
+
+    internal static bool IsFluentThemeEnabled
+    {
+        get
+        {
+            if(Application.Current == null) return false;
```

Comment:
Nit: Space after `if` + `return` on new line for readbility or just `Application.Current?.ThemeMode != ThemeMode.None;` possibly?

Reply:
I left it that way for readability.

Reply:
I've only just noticed that you use it all the way through the code, yeah, then it's fine.